dolphinscheduler-common/src/main/java/org/apache/dolphinscheduler/common/config/IYamlDelegate.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.dolphinscheduler.common.config;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Function;
+
+public interface IYamlDelegate {
```

Comment:
You don't need to create this interface, we already have a `IPropertyDelegate` interface, the yaml implementation can directly implement this.

Reply:
I think maybe we can not leverage `IPropertyDelegate`, configuration becomes nesting just like JSONObject in jackson, so I do not choose the design when I writing this part.

Reply:
```
s3:
    access-key: aaa
    access-card: false
```
for example. if you wants to get access key in s3 part. you should firstly get mapping object of s3. in s3 mappig object you can get access-key as string instead of directly getting `s3.access-key` like properties.

Reply:
> ```
> s3:
>     access-key: aaa
>     access-card: false
> ```
> 
> for example. if you wants to get access key in s3 part. you should firstly get mapping object of s3. in s3 mappig object you can get access-key as string instead of directly getting `s3.access-key` like properties.

We should wrap this into `S3_ACCESS_KEY`, then this should same with properties, and it can be overridden by env.

Reply:
per offline sync. i will reference `YamlPropertiesFactoryBean` to implement another